static/js/src/advantage/manage-subscription.js

```diff
@@ -1,93 +1,259 @@
-import { resizeContract, cancelContract } from "./contracts-api.js";
+import {
+  resizeContract,
+  cancelContract,
+  getPurchase,
+} from "./contracts-api.js";
+
+const stripe = window.Stripe(window.stripePublishableKey);
+
+const getMessage = (code, default_message) => {
+  const map = {
+    resizing_machines_success: "Resizing successful! Reloading page...",
+    cancelling_subscription_success:
+      "Subscription cancelled! Reloading page...",
+    resizing_machines_fail:
+      "<strong>Payment method:</strong> There was an error with the payment please <a href='/advantage/payment-methods'>update your payment methods</a> to fix it.",
+    subscription_missing:
+      "<strong>Cancelling subscription failed:</strong> It could be that you have a pending payment that is blocking this action. Contact <a class='p-notification__action' href='https://ubuntu.com/contact-us'>Canonical sales</a> if the problem persists.",
+    cancelling_subscription_failed:
+      "<strong>Cancelling subscription failed:</strong> Contact <a class='p-notification__action' href='https://ubuntu.com/contact-us'>Canonical sales</a> if the problem persists.",
+    pending_purchase:
+      "<strong>Error:</strong> You already have a pending purchase. Please go to <a href='/advantage/payment-methods'>payment methods</a> to fix it. Reloading page...",
+    unknown_error:
+      "<strong>Unknown error:</strong> Contact <a class='p-notification__action' href='https://ubuntu.com/contact-us'>Canonical sales</a> if the problem persists.",
+  };
```

Comment:
@wgx @elioqoshi please double-check the copy is fitting our tone

Reply:
done